docs/specs/balance-fetcher.md

```diff
@@ -0,0 +1,311 @@
+# Balance Fetcher Technical Specification
+
+## 1. Overview
+
+The Balance Fetcher is a critical component in the AEP Fee Calculator system that retrieves and stores historical balance information for reward distributors on the Arbitrum Nova network. It queries blockchain state at specific end-of-day blocks to capture point-in-time balance snapshots necessary for accurate fee calculations.
+
+## 2. Purpose
+
+### Primary Objectives
+
+- Retrieve end-of-day ETH balances for all tracked reward distributors
```

Comment:
No, only does it for one distributor - a new instance would be created per distributor